pkg/repo/mysql/message.go

```diff
@@ -99,36 +100,58 @@ ON DUPLICATE KEY UPDATE
 	return nil
 }
 
-func (r *Repo) LoadMessages(ctx context.Context, serviceID uuid.UUID, language language.Tag) (*model.Messages, error) {
-	messages := &model.Messages{Language: language}
-
-	rows, err := r.db.QueryContext(
-		ctx,
-		`SELECT mm.id, mm.message, mm.description, mm.fuzzy
-FROM message_message mm
-JOIN message m ON m.id = mm.message_id
-WHERE m.service_id = UUID_TO_BIN(?) AND m.language = ?`,
-		serviceID,
-		language.String(),
-	)
+func (r *Repo) LoadMessages(ctx context.Context, serviceID uuid.UUID, opts common.LoadMessagesOpts,
+) ([]model.Messages, error) {
+	rows, err := sq.RunWith(r.db).
+		Select("mm.id, mm.message, mm.description, mm.fuzzy, m.language").
+		From("message_message mm").Join("message m ON m.id = mm.message_id").
+		Where("m.service_id = UUID_TO_BIN(?)", serviceID).
+		Where(make(eb).in("m.language", langTagsToStringSlice(opts.FilterLanguages)).eq()).
+		QueryContext(ctx)
 	if err != nil {
 		return nil, fmt.Errorf("repo: query messages: %w", err)
 	}
 
 	defer rows.Close()
 
+	messagesLookup := make(map[string][]model.Message)
+
 	for rows.Next() {
-		var m model.Message
-		if err := rows.Scan(&m.ID, &m.Message, &m.Description, &m.Fuzzy); err != nil {
+		var (
+			msg  model.Message
+			lang string
+		)
+
+		if err := rows.Scan(&msg.ID, &msg.Message, &msg.Description, &msg.Fuzzy, &lang); err != nil {
 			return nil, fmt.Errorf("repo: scan message: %w", err)
 		}
 
-		messages.Messages = append(messages.Messages, m)
+		messagesLookup[lang] = append(messagesLookup[lang], msg)
 	}
 
 	if err := rows.Err(); err != nil {
 		return nil, fmt.Errorf("repo: scan messages: %w", err)
 	}
 
+	messages := make([]model.Messages, 0, len(messagesLookup))
+
+	for langTag, msgs := range messagesLookup {
+		messages = append(messages, model.Messages{
+			Language: language.MustParse(langTag),
```

Comment:
error handling instead of panic e.g. MustParse -> Parse.

for this case i think you could define struct that implements Scanner and Valuer, to get rid off manual parsing in all repo functions

e.g.

```go
type lang struct {
  language.Tag
}

func (l lang) Value() (driver.Value, error) {
...
}

func (l *lang) Scan(value interface{}) error {
...
}
```

Reply:
In the end, both ways are right. The probability of it happening is very low, we always ensure we keep clean and sanitized data in our system. Users still can edit data directly in the database which happens a lot.

`driver.Valuer` and `sql.Scanner` would be nicer, the same implementation across multiple SQL repo implementations. Leaving it as tech debt.